Dockerfile

```diff
@@ -1,25 +1,77 @@
-# from https://hub.docker.com/_/ubuntu/tags?page=1&name=22.04
-FROM ubuntu:22.04@sha256:bcc511d82482900604524a8e8d64bf4c53b2461868dac55f4d04d660e61983cb
+FROM ghcr.io/actions/actions-runner:2.313.0 AS base
+USER root
 
-WORKDIR /
+RUN apt-get update \
+    && apt-get install -y curl jq \
+    && apt-get -y install curl git vim \
+    && apt-get -y install zip unzip \
+    && apt-get -y install ca-certificates curl wget apt-transport-https lsb-release gnupg \
+    && apt-get satisfy "python3-pip (<= 22.1)" -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/
 
-COPY dockerfile-setup.sh dockerfile-setup.sh
-RUN bash dockerfile-setup.sh
+FROM base AS deps
+RUN bash bin/installdependencies.sh \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
-COPY github-runner-entrypoint.sh /github-runner-entrypoint.sh
-RUN chmod +x /github-runner-entrypoint.sh
+FROM deps AS deps-az
+RUN curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null
+RUN AZ_REPO=$(lsb_release -cs) && \
+    echo "deb [arch=$(dpkg --print-architecture)] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | tee /etc/apt/sources.list.d/azure-cli.list
+RUN apt-get update && \
+    apt-get -y install azure-cli \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
-# changed user to avoid root user
-USER github
+RUN az config set extension.use_dynamic_install=yes_without_prompt
 
-RUN whoami && \
-  az --version && \
-  kubectl --help && \
-  kubelogin --version && \
-  helm --help && \
-  yq --version && \
-  node --version && \
-  npm --version && \
-  yarn --version
+# 4) 1.81GB
 
-ENTRYPOINT ["/github-runner-entrypoint.sh"]
+# rivedere
+FROM deps-az AS deps-node-yarn
+RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/trusted.gpg.d/nodesource.gpg
+ENV NODE_MAJOR_VERSION="${ENV_NODE_MAJOR_VERSION:-20}"
```

Comment:
why install node? we have a lot of projects with different node versions so maybe it's better not install it by default (yarn it's ok).
Probably we introduce node in other PR not related to this.